core/src/main/java/org/geysermc/geyser/translator/protocol/bedrock/entity/player/BedrockInteractTranslator.java

```diff
@@ -74,7 +74,7 @@ public void translate(GeyserSession session, InteractPacket packet) {
             case LEAVE_VEHICLE:
                 ServerboundPlayerCommandPacket sneakPacket = new ServerboundPlayerCommandPacket((int) entity.getEntityId(), PlayerState.START_SNEAKING);
                 session.sendDownstreamPacket(sneakPacket);
-                session.setRidingVehicleEntity(null);
+                session.getPlayerEntity().setVehicle(null);
```

Comment:
Referencing vanilla code, this may not be intended behavior - Java just sends that the shift (or equivalent) key is being pressed. I think we have existing issues about Bedrock prematurely dismounting entities on its own - wonder if we can work around that.

Reply:
I guess our only option is to resend the entity link to force the player back onto vehicle.

Reply:
Fun stuff. We should just decide if we need to remove this line or not - we can save workarounds for another time.

Reply:
Since you said it's server authoritative, I'll just remove the line.